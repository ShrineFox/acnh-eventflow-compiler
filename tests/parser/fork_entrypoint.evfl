flow Test():
    fork:
        branch:
entrypoint EP1:
            TestActor1.Action0()
        branch:
            TestActor2.Action0()
entrypoint EP2:
            TestActor1.Action0()
entrypoint EP3:
    TestActor1.Action0()

