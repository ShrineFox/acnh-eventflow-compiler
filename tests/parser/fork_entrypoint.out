RootNode[name=Test, vardefs=[], out_edges=[Event8]]
ActionNode[name=Event0, action=EventFlowActionAction0(), params={}, out_edges=[]]
ActionNode[name=Event3, action=EventFlowActionAction0(), params={}, out_edges=[Event5]]
ActionNode[name=Event5, action=EventFlowActionAction0(), params={}, out_edges=[]]
ForkNode[name=Event8, join_node=Event9, out_edges=[Event0, Event3, Event9]]
JoinNode[name=Event9, out_edges=[Event11]]
ActionNode[name=Event11, action=EventFlowActionAction0(), params={}, out_edges=[]]

